client/state/login/actions.js

```diff
@@ -52,6 +53,57 @@ import wpcom from 'lib/wp';
 import { addLocaleToWpcomUrl, getLocaleSlug } from 'lib/i18n-utils';
 import { recordTracksEventWithClientId as recordTracksEvent } from 'state/analytics/actions';
 
+/**
+ * Creates a promise that will be rejected after a given timeout
+ *
+ * @param {int} ms amount of milliseconds till reject the promise
+ * @returns {Promise} a promise that will be rejected after ms milliseconds
+ */
+export const createTimingOutPromise = ms =>
+	new Promise( ( _, reject ) => {
+		setTimeout( () => reject( new Error( `timeout of ${ ms } reached` ) ), ms );
+	} );
+
+/**
+ * Makes a request to a given link in an iframe
+ *
+ * @param {string} loginLink the login link to load
+ * @param {int} requestTimeout amount of time to allow the link to load, default 25s
+ * @returns {Promise} a promise that will be resolved if the link was successfully loaded
+ */
+export const makeRemoteLoginRequest = ( loginLink, requestTimeout = 25000 ) => {
+	let iframe;
+	const iframeLoadPromise = new Promise( resolve => {
+		iframe = document.createElement( 'iframe' );
+		iframe.style.display = 'none';
+		iframe.setAttribute( 'scrolling', 'no' );
+		iframe.onload = resolve;
+		iframe.src = loginLink;
+		document.body.appendChild( iframe );
+	} );
+
+	return Promise.race( [ iframeLoadPromise, createTimingOutPromise( requestTimeout ) ] ).finally(
+		() => {
+			iframe.parentElement.removeChild( iframe );
+		}
+	);
```

Comment:
can we make sure that this works in IE? I noted the `finally()` isn't natively supported there and I wasn't sure if our transform steps are able to transform this

Reply:
we polyfill Promise, so it should be fine

Reply:
thanks @blowery

Reply:
guess it would have been useful to _test_ this to _make sure that it works_ 😉